cli/internal/lockfile/pnpm_lockfile.go

```diff
@@ -0,0 +1,234 @@
+	OptionalDependencies map[string]string         `yaml:"optionalDependencies,omitempty"`
+	DevDependencies      map[string]string         `yaml:"devDependencies,omitempty"`
+	DependenciesMeta     map[string]DependencyMeta `yaml:"dependenciesMeta,omitempty"`
+	PublishDirectory     string                    `yaml:"publishDirectory,omitempty"`
+}
+
+// PackageSnapshot Snapshot used to represent a package in the packages setion
+type PackageSnapshot struct {
+	ID string `yaml:"id,omitempty"`
+
+	Resolution PackageResolution `yaml:"resolution,flow"`
+	Engines    struct {
+		Node string `yaml:"node"`
+		NPM  string `yaml:"npm,omitempty"`
+	} `yaml:"engines,omitempty,flow"`
+	CPU           []string `yaml:"cpu,omitempty,flow"`
+	Os            []string `yaml:"os,omitempty,flow"`
+	HasBin        bool     `yaml:"hasBin,omitempty"`
+	RequiresBuild bool     `yaml:"requiresBuild,omitempty"`
+
+	PeerDependencies     map[string]string `yaml:"peerDependencies,omitempty"`
+	PeerDependenciesMeta map[string]struct {
+		Optional bool `yaml:"optional"`
+	} `yaml:"peerDependenciesMeta,omitempty"`
+	Dependencies         map[string]string `yaml:"dependencies,omitempty"`
+	OptionalDependencies map[string]string `yaml:"optionalDependencies,omitempty"`
+	TransitivePeerDeps   []string          `yaml:"transitivePeerDependencies,omitempty"`
+	BundledDependencies  []string          `yaml:"bundledDependencies,omitempty"`
+
+	Dev      bool `yaml:"dev"`
+	Optional bool `yaml:"optional,omitempty"`
+	Patched  bool `yaml:"patched,omitempty"`
+	Prepare  bool `yaml:"prepare,omitempty"`
+
+	// only needed for packages that aren't in npm
+	Name    string `yaml:"name,omitempty"`
+	Version string `yaml:"version,omitempty"`
+
+	LibC       []string `yaml:"libc,omitempty"`
+	Deprecated string   `yaml:"deprecated,omitempty"`
+}
+
+// PackageResolution Various resolution strategies for packages
+type PackageResolution struct {
+	Type string `yaml:"type,omitempty"`
+	// For npm or tarball
+	Integrity string `yaml:"integrity,omitempty"`
+
+	// For tarball
+	Tarball string `yaml:"tarball,omitempty"`
+
+	// For local directory
+	Dir string `yaml:"directory,omitempty"`
+
+	// For git repo
+	Repo   string `yaml:"repo,omitempty"`
+	Commit string `yaml:"commit,omitempty"`
+}
+
+// PatchFile represent a patch applied to a package
+type PatchFile struct {
+	Path string `yaml:"path"`
+	Hash string `yaml:"hash"`
+}
+
+func isSupportedVersion(version float32) error {
+	supportedVersions := []float32{5.3, 5.4}
+	for _, supportedVersion := range supportedVersions {
+		if version == supportedVersion {
+			return nil
+		}
+	}
+	return errors.Errorf("Unable to generate pnpm-lock.yaml with lockfileVersion: %f. Supported lockfile versions are %v", version, supportedVersions)
+}
+
+// DependencyMeta metadata for dependencies
+type DependencyMeta struct {
+	Injected bool   `yaml:"injected,omitempty"`
+	Node     string `yaml:"node,omitempty"`
+	Patch    string `yaml:"string,omitempty"`
+}
```

Comment:
I was trying to follow along on where this came from; can you add a comment here? (I still haven't figured it out.)

Reply:
`DependenciesMeta` is the upstream name so I couldn't find it. Can we rename? (Found it.)